src/whenIdle.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * whenIdle executes the given callback when the browser is 'idle'
+ *
+ * @param callback Fired when requestIdleCallback runs or, if requestIdleCallback is not available, after 300ms
+ */
+export const whenIdle = (callback: () => void): void => {
+	if ('requestIdleCallback' in window) {
+		window.requestIdleCallback(callback, { timeout: 500 });
```

Comment:
There is a case to be made for allowing the timeout to be overridable as part of the API (defaulting to 500ms). What do you think?

Reply:
`options`!